common/djangoapps/student/tests/test_roles.py

```diff
@@ -174,16 +176,13 @@ def setUp(self):
 
     @ddt.data(*ROLES)
     @ddt.unpack
-    def test_only_in_role(self, role, target):
+    def test_only_in_role(self, role, target, test_id):
         role.add_users(self.user)
         cache = RoleCache(self.user)
         assert cache.has_role(*target)
 
-        for other_role, other_target in self.ROLES:
-            if other_role == role:
-                continue
-
-            assert not cache.has_role(*other_target)
+        for _, other_target, other_test_id in self.ROLES:
+            assert cache.has_role(*other_target) == (test_id == other_test_id)
```

Comment:
@shadinaif I presume I'm very familiar with the issue being presented here, but the test still isn't understanble.

I think the real cause is that the original `test_only_in_role` function was cryptic.

In such case I would support creating a new test case e.g. `test_only_in_role_case_insensitive` so it doesn't have for-loop or at least it doesn't have to be so generic and becomes hard to understand.